pkg/resource/obtenant_manager.go

```diff
@@ -0,0 +1,726 @@
+	}
+	poolList, err := oceanbaseOperationManager.GetPoolList()
+	if err != nil {
+		return nil, errors.Wrap(err, "Get sql error when get pool list")
+	}
+
+	poolIdMap := make(map[int64]string, 0)
+	for _, pool := range poolList {
+		if pool.TenantID.Valid && pool.TenantID.Int64 == tenantID {
+			poolIdMap[pool.ResourcePoolID] = pool.Name
+		}
+	}
+	zoneMap := make(map[string]string, 0)
+	unitList, err := oceanbaseOperationManager.GetUnitList()
+	if err != nil {
+		return nil, errors.Wrap(err, "Get sql error when get unit list")
+	}
+	for _, unit := range unitList {
+		if poolIdMap[unit.ResourcePoolID] != "" && zoneMap[unit.Zone] == "" {
+			zoneMap[unit.Zone] = unit.Zone
+		}
+	}
+	for k, _ := range zoneMap {
+		zoneList = append(zoneList, k)
+	}
+	return zoneList, nil
+}
+
+
+
+func (m *OBTenantManager) GenerateStatusUnitNumMap(zones []v1alpha1.ResourcePoolSpec) (map[string]int, error) {
+	unitNumMap := make(map[string]int, 0)
+	oceanbaseOperationManager, err := m.getOceanbaseOperationManager()
+	if err != nil {
+		return unitNumMap, errors.Wrap(err, "Get Sql Operator Error When Building Resource Unit From DB")
+	}
+	poolList, err := oceanbaseOperationManager.GetPoolList()
+	if err != nil {
+		return unitNumMap, errors.Wrap(err, "Get sql error when get pool list")
+	}
+	for _, zone := range zones {
+		poolName := m.GeneratePoolName(zone.ZoneList)
+		for _, pool := range poolList {
+			if pool.Name == poolName {
+				unitNumMap[zone.ZoneList] = int(pool.UnitNum)
+			}
+		}
+	}
+	return unitNumMap, nil
+}
+
+func (m *OBTenantManager) BuildUnitConfigV4FromDB(zone string, tenantID int64) (v1alpha1.UnitConfig, error) {
+	var unitConfig v1alpha1.UnitConfig
+	oceanbaseOperationManager, err := m.getOceanbaseOperationManager()
+	if err != nil {
+		return unitConfig, errors.Wrap(err, "Get Sql Operator Error When Building Resource Unit From DB")
+	}
+	unitList, err := oceanbaseOperationManager.GetUnitList()
+	if err != nil {
+		return unitConfig, errors.Wrap(err, "Get sql error when get unit list")
+	}
+	poolList, err := oceanbaseOperationManager.GetPoolList()
+	if err != nil {
+		return unitConfig, errors.Wrap(err, "Get sql error when get pool list")
+	}
+	unitConfigList, err := oceanbaseOperationManager.GetUnitConfigV4List()
+	if err != nil {
+		return unitConfig, errors.Wrap(err, "Get sql error when get unit config list")
+	}
+	var resourcePoolIDList []int
+	for _, unit := range unitList {
+		if unit.Zone == zone {
+			resourcePoolIDList = append(resourcePoolIDList, int(unit.ResourcePoolID))
+		}
+	}
+	for _, pool := range poolList {
+		for _, resourcePoolID := range resourcePoolIDList {
+			if resourcePoolID == int(pool.ResourcePoolID) && pool.TenantID.Valid && pool.TenantID.Int64 == tenantID {
+				for _, unitConifg := range unitConfigList {
+					if unitConifg.UnitConfigID == pool.UnitConfigID {
+						unitConfig.MaxCPU = kuberesource.MustParse(strconv.FormatFloat(unitConifg.MaxCPU, 'f', -1, 64))
```

Comment:
MustParse will panic if there's something wrong, will cause the whole process restart

Reply:
Has been modified to the more secure parse method.